modules/core/src/test/java/org/apache/ignite/internal/processors/performancestatistics/PerformanceStatisticsSelfTest.java

```diff
@@ -0,0 +1,290 @@
+        @Override public Object process(MutableEntry<Object, Object> entry, Object... arguments)
+            throws EntryProcessorException {
+            return null;
+        }
+    };
+
+    /** Test cache entry processor. */
+    private static final CacheEntryProcessor<Object, Object, Object> CACHE_ENTRY_PROC =
+        new CacheEntryProcessor<Object, Object, Object>() {
+        @Override public Object process(MutableEntry<Object, Object> entry, Object... arguments)
+            throws EntryProcessorException {
+            return null;
+        }
+    };
+
+    /** Cache entry count. */
+    private static final int ENTRY_COUNT = 100;
+
+    /** Ignite. */
+    private static IgniteEx ignite;
+
+    /** Test cache. */
+    private static IgniteCache<Object, Object> cache;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setCacheConfiguration(defaultCacheConfiguration());
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        log = new ListeningTestLogger(GridAbstractTest.log);
+
+        ignite = startGrid(0);
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        cache = ignite.cache(DEFAULT_CACHE_NAME);
+
+        for (int i = 0; i < ENTRY_COUNT; i++)
+            cache.put(i, i);
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testCompute() throws Exception {
+        String testTaskName = "testTask";
+        int executions = 5;
+
+        startCollectStatistics();
+
+        IgniteRunnable task = new IgniteRunnable() {
+            @Override public void run() {
+                // No-op.
+            }
+        };
+
+        for (int i = 0; i < executions; i++)
+            ignite.compute().withName(testTaskName).run(task);
+
+        HashMap<IgniteUuid, Integer> sessions = new HashMap<>();
+        AtomicInteger tasks = new AtomicInteger();
+        AtomicInteger jobs = new AtomicInteger();
+
+        stopCollectStatisticsAndRead(new TestHandler() {
+            @Override public void task(UUID nodeId, IgniteUuid sesId, String taskName, long startTime, long duration,
+                int affPartId) {
+                sessions.compute(sesId, (uuid, val) -> val == null ? 1 : ++val);
+
+                tasks.incrementAndGet();
+
+                assertEquals(ignite.context().localNodeId(), nodeId);
+                assertEquals(testTaskName, taskName);
+                assertTrue(startTime > 0);
+                assertTrue(duration >= 0);
+                assertEquals(-1, affPartId);
+            }
+
+            @Override public void job(UUID nodeId, IgniteUuid sesId, long queuedTime, long startTime, long duration,
+            boolean timedOut) {
+                sessions.compute(sesId, (uuid, val) -> val == null ? 1 : ++val);
```

Comment:
`val` should be renamed to `cnt` here and below.

Reply:
Done.